internal/util/k8s/persistentvolumes.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2025 The CephCSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetPersistentVolume returns the PersistentVolume object for the given pvName.
+func GetPersistentVolume(pvName string, ignoreNotFound bool) (*corev1.PersistentVolume, error) {
+	client, err := NewK8sClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to Kubernetes: %w", err)
+	}
+
+	pv, err := client.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
```

Comment:
Eventually all `context.TODO()` should be replaced by the actual context for the gRPC call. Not all interactions with Kubernetes may come through a gRPC call, and those should create their own context (as high in the stack as possible).

Just an other cleanup task for the future :smile: